### PR TITLE
control-service: java 17

### PR DESCRIPTION
**Pull request by murphp15**
# Why
Java 17 is more secure and needed for a lot of modern libraries. 
More details can be seen here: https://github.com/vmware/versatile-data-kit/issues/1440

# What
When using java 17 `time.now()` returns nanosecond accuracy. 
This causes problems for us because we only store micro second accuracy in the database.
To overcome this for a lot of tests we use timestamps which are accurate to micro seconds only. 

# How has this been tested?
Unit and integration tests

closes: https://github.com/vmware/versatile-data-kit/issues/1440 
Signed-off-by: murphp15 <murphp15@tcd.ie>

### CODE REVIEW

File: projects/control-service/cicd/.gitlab-ci.yml [hunk 1]
```diff
@@ -22,24 +22,25 @@
   - projects/control-service/projects/helm_charts/pipelines-control-service/**/*
 
 .control_service_base_build:
-  image: docker:19.03.15
+  image: docker:20.10.21
   variables:
```
Comment (ivakoleva) [inline projects/control-service/cicd/.gitlab-ci.yml:26]: The pipeline seems to fail during:
```
[32;1m$ pip3 install --upgrade pip && pip3 install awscli[0;m
Traceback (most recent call last):
  File "/usr/bin/pip3", line 33, in <module>
    sys.exit(load_entry_point('pip==22.3.1', 'console_scripts', 'pip3')())
  File "/usr/bin/pip3", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 969, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 548, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for pip
```
In comparison to a successful one that looks like:
```
32;1m$ pip3 install --upgrade pip && pip3 install awscli[0;m
Looking in indexes: https://pypi.org/simple, https://pypi.org/simple
Requirement already satisfied: pip in /usr/lib/python3.8/site-packages (20.3.4)
Collecting pip
  Downloading pip-22.3.1-py3-none-any.whl (2.1 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.3.4
    Uninstalling pip-20.3.4:
      Successfully uninstalled pip-20.3.4
Successfully installed pip-22.3.1
```

I would suggest to test if unblocked by some of the solutions mentioned in  https://stackoverflow.com/questions/54552367/pip-cannot-find-metadata-file-environmenterror
For example force reinstall

**Review by ivakoleva — CHANGES_REQUESTED** [review]

**Comment by mivanov1988** [comment]
Great, thanks for dealing with this!

File: projects/control-service/projects/pipelines_control_service/Dockerfile [hunk 2]
```diff
@@ -4,15 +4,15 @@
 # Builder stage
 # This stage is implemented like in the guide, but it may be more efficient to
 # have it in the gradle build process outside docker.
-FROM adoptopenjdk:11-jre-hotspot as builder
+FROM eclipse-temurin:17 as builder
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/Dockerfile:7]: What are the differences between adoptopenjdk and eclipse-temurin?

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/graphql/it/GraphQLExecutionsLogsUrlIT.java [hunk 3]
```diff
@@ -132,7 +133,7 @@ public void testExecutions_filterByStartTimeGte() throws Exception {
                     "{"
                         + "\"filter\": {"
                         + "      \"startTimeGte\": \""
-                        + dataJobExecution3.getStartTime()
+                        + getTimeAccurateToMicroSecond()
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/graphql/it/GraphQLExecutionsLogsUrlIT.java:136]: shouldn't you use a method that accepts dataJobExecution3.getStartTime()?

Comment (murphp15) [reply projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/graphql/it/GraphQLExecutionsLogsUrlIT.java:136]: strange let me look into this.

File: projects/control-service/projects/pipelines_control_service/Dockerfile (hunk 2 above)
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/Dockerfile:7]: Thanks!

**Review by mivanov1988 — APPROVED** [review]

**Review by tozka — APPROVED** [review]

File: projects/control-service/cicd/.gitlab-ci.yml (hunk 1 above)
Comment (murphp15) [reply projects/control-service/cicd/.gitlab-ci.yml:33]: Done

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/RepositoryUtil.java [hunk 4]
```diff
@@ -151,4 +152,14 @@ public static DataJobExecution createDataJobExecution(
...
     return jobExecutionRepository.save(expectedJobExecution);
   }
+
+  /**
+   * at the database level we only store date-times accurate to the microsecond. Like wise in older
+   * versions of java .now() returned timestamps accurate to micro-seconds. In newer versions of
+   * java .now() gives nano-second precision and it causes tests written before we adopted that java
+   * version to fail.
+   */
+  public static OffsetDateTime getTimeAccurateToMicroSecond() {
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/RepositoryUtil.java:162]: I find the documentation great. My only concern is with duplicated methods
with JobExecutionUtil

**Review by ivakoleva — APPROVED** [review]
LGTM
nit: maybe link at major diff 11->17 transition in case you've looked for it

**Pull request merged**